### PR TITLE
BUG: Do not ignore empty tuple of strides in ndarray.__new__

### PR DESCRIPTION
**Pull request by eric-wieser**
Previously this was treated as though an empty set of strides were passed.

Now this is treated as a request for `.strides == ()`.

---

First commit is shared with #15881

### CODE REVIEW

File: numpy/core/src/multiarray/arrayobject.c [hunk 1]
```diff
@@ -1645,7 +1645,7 @@ array_new(PyTypeObject *subtype, PyObject *args, PyObject *kwds)
                                      PyArray_BufferConverter,
                                      &buffer,
                                      &offset,
-                                     &PyArray_IntpConverter,
+                                     &PyArray_OptionalIntpConverter,
```
Comment (mattip) [inline numpy/core/src/multiarray/arrayobject.c]: error: ‘PyArray_OptionalIntpConverter’ undeclared

Comment (eric-wieser) [reply numpy/core/src/multiarray/arrayobject.c]: Any idea why, given it comes from the same file as the previous symbol here?

Comment (seberg) [reply numpy/core/src/multiarray/arrayobject.c]: The other symbol is part of the Public API so declared elsewhere.

**Comment by mattip** [comment]
Since this is a change in behaviour from 1.17, maybe it should have a release note?

**Comment by eric-wieser** [comment]
Can't hurt, added a release note

**Comment by mattip** [comment]
Thanks @eric-wieser

**Pull request merged**